src/object/tcb.c

```diff
@@ -1604,9 +1642,15 @@ exception_t decodeDomainInvocation(word_t invLabel, word_t length, word_t *buffe
         current_syscall_error.invalidArgumentNumber = 1;
         return EXCEPTION_SYSCALL_ERROR;
     }
+    tcb = TCB_PTR(cap_thread_cap_get_capTCBPtr(tcap));
+    return invokeDomainSetSet(tcb, domain);
+}
 
+exception_t invokeDomainSetSet(tcb_t *tcb, dom_t domain)
+{
     setThreadState(NODE_STATE(ksCurThread), ThreadState_Restart);
```

Comment:
Ah, that took a while to get back up to and to remember. To follow the structure of other decode/invokes and to allow us to reuse existing proofs, this `setThreadState` should still be in `decodeDomainInvocation, before the call of `invokeDomainSetSet`.